src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -168,6 +177,37 @@ internal class TracerProviderSdk : TracerProvider
 
         internal Resource Resource { get; }
 
+        internal static bool CheckVersions(string activitySourceVersion, Source source)
+        {
+            if (string.IsNullOrEmpty(activitySourceVersion))
+            {
+                return true;
+            }
+
+            if (!Version.TryParse(activitySourceVersion, out Version currentVersion))
+            {
+                return true;
+            }
+
+            if (source.MinVersion == null && source.MaxVersion == null)
+            {
+                return true;
+            }
+
+            bool validation = false;
+            if (source.MinVersion != null)
+            {
+                validation = source.MinVersion <= currentVersion;
+            }
+
+            if (source.MaxVersion != null)
+            {
+                validation = source.MaxVersion >= currentVersion;
+            }
+
+            return validation;
```

Comment:
`return (source.MinVersion <= currentVersion && currentVersion <=source.MaxVersion)` - this is what you want right?

Reply:
if we do this, we can receive a null reference when doing the compare.
minver = null
current ver = 1.0.0
maxver = 2.0.0

in this test, your snippet would throw an exception.

Reply:
I meant conceptually, not the exact code 😄 
```
            bool minVersionOkay = true;
            if (source.MinVersion != null)
            {
                minVersionOkay = source.MinVersion <= currentVersion;
            }

            bool maxVersionOkay = true;
            if (source.MaxVersion != null)
            {
                maxVersionOkay = source.MaxVersion >= currentVersion;
            }

            return maxVersionOkay && mixVersionOkay ;
```

Reply:
yes, thats right.